app/src/main/java/com/alphawallet/app/ui/widget/adapter/TestNetHorizontalListAdapter.java

```diff
@@ -0,0 +1,103 @@
+package com.alphawallet.app.ui.widget.adapter;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.alphawallet.app.R;
+import com.alphawallet.app.entity.tokens.Token;
+import com.alphawallet.app.entity.tokens.TokenCardMeta;
+import com.alphawallet.app.repository.EthereumNetworkRepository;
+import com.alphawallet.app.service.AssetDefinitionService;
+import com.alphawallet.app.service.TokensService;
+import com.alphawallet.app.widget.TokenIcon;
+
+import java.math.BigInteger;
+import java.util.Locale;
+
+public class TestNetHorizontalListAdapter extends RecyclerView.Adapter<TestNetHorizontalListAdapter.ViewHolder>
+{
+    private final Token[] tokens;
+    private final Context context; //TODO - JB: should be final
+
+    //TODO - JB: populate the token list using this method:
+    // - from tokensService get the tokenFilter with getNetworkFilters()
+    // - loop through this list and check for non-zero balance testnet (using getTokenOrBase(chainId, tokensService.getCurrentAddress()) )
+    // - send list of tokens below but use Token[] instead of TokenCardMeta[]. Now you won't need TokensService or AssetDefinitionService
+
+    public TestNetHorizontalListAdapter(Token[] tokens, Context context)
+    {
+        this.tokens = tokens;
+        this.context = context;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
+    {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.item_horizontal_testnet_list, parent, false);
+        return new TestNetHorizontalListAdapter.ViewHolder(view);
+    }
+
+    //TODO - JB: Fix formatting (no whitespace at top of function, fix tabbing)
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position)
+    {
+        Token token = tokens[position];
+        holder.tokenIcon.clearLoad();
+        try
+        {
+            String coinBalance = token.getStringBalanceForUI(4);
+            if (!TextUtils.isEmpty(coinBalance))
+            {
+                String symbol = token.getSymbol().substring(0, Math.min(token.getSymbol().length(), 5))
```

Comment:
You can extract a method `getXXSymbol()` into `Token`, and write a unit test for it.

Reply:
@keval-finimble we actually have a standard method for displaying symbol in these conditions: checkout the source for```setupTextIcon``` in ```class TokenIcon```